### PR TITLE
Add emp artifact effect

### PR DESCRIPTION
**Pull request by Slava0135**
## About the PR

**Media**

Peek 2023-03-07 22-07

**Changelog**

:cl:
- add: added emp artifact effect.

### CODE REVIEW

File: Content.Server/Xenoarchaeology/XenoArtifacts/Effects/Components/EmpArtifactComponent.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+namespace Content.Server.Xenoarchaeology.XenoArtifacts.Effects.Components;
+
+/// <summary>
+/// Artifact that EMP
+/// </summary>
+[RegisterComponent]
```
Comment (EmoGarbage404) [inline Content.Server/Xenoarchaeology/XenoArtifacts/Effects/Components/EmpArtifactComponent.cs:6]: access

File: Content.Server/Xenoarchaeology/XenoArtifacts/Effects/Components/EmpArtifactComponent.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+public sealed class EmpArtifactComponent : Component
+{
+    [DataField("range")]
```
Comment (EmoGarbage404) [inline Content.Server/Xenoarchaeology/XenoArtifacts/Effects/Components/EmpArtifactComponent.cs:9]: public fields in components need datafields. Yes i know it's obvious; it still needs to be done.

File: Resources/Prototypes/XenoArch/Effects/normal_effects.yml [hunk 3]
```diff
@@ -495,3 +495,10 @@
     maxSpawns: 1
     spawns:
     - id: Singularity
+
+- type: artifactEffect
+  id: EffectEmp
+  targetDepth: 2
```
Comment (EmoGarbage404) [inline Resources/Prototypes/XenoArch/Effects/normal_effects.yml:501]: all of the effects are loosely sorted from low to high depth. At the very least, put this one in the section with the other depth 2 artifacts. Alternatively, bump it up to depth 3.

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

File: Content.Server/Xenoarchaeology/XenoArtifacts/Effects/Components/EmpArtifactComponent.cs (hunk 2 above)
Comment (Slava0135) [reply Content.Server/Xenoarchaeology/XenoArtifacts/Effects/Components/EmpArtifactComponent.cs:9]: Fun fact: I took it from ignite artifact, so you may want to fix it there

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**